libs/remix-ui/editor-context-view/src/lib/remix-ui-editor-context-view.tsx

```diff
@@ -0,0 +1,178 @@
+import React, { useEffect, useState, useRef } from 'react' // eslint-disable-line
+import { sourceMappingDecoder } from '@remix-project/remix-debug'
+
+import './remix-ui-editor-context-view.css'
+
+/* eslint-disable-next-line */
+
+export type astNode = {
+  name: string,
+  id: number,
+  children: Array<any>,
+  typeDescriptions: any,
+  nodeType: String,
+  src: any,
+  nodeId: any,
+  position: any
+}
+
+export type onContextListenerChangedListener = (nodes: Array<astNode>) => void
+
+export type gasEstimationType = {
+  executionCost: string,
+  codeDepositCost: string
+}
+export interface RemixUiEditorContextViewProps {
+  hide: boolean,
+  gotoLine: (line: number, column: number) => void,
+  openFile: (fileName: string) => void,
+  getLastCompilationResult: () => any,
+  offsetToLineColumn: (position: any, file: any, sources: any, asts: any) => any,
+  getCurrentFileName: () => String
+  onContextListenerChanged: (listener: onContextListenerChangedListener) => void
+  referencesOf: (nodes: astNode) => Array<astNode>
+  getActiveHighlights: () => Array<astNode>
+  gasEstimation: (node: astNode) => gasEstimationType
+  declarationOf: (node: astNode) => astNode
+}
+
+function isDefinition (node: any) {
+  return node.nodeType === 'ContractDefinition' ||
+  node.nodeType === 'FunctionDefinition' ||
+  node.nodeType === 'ModifierDefinition' ||
+  node.nodeType === 'VariableDeclaration' ||
+  node.nodeType === 'StructDefinition' ||
+  node.nodeType === 'EventDefinition'
+}
+
+
+
+type nullableAstNode = astNode | null
+
+export function RemixUiEditorContextView (props: RemixUiEditorContextViewProps) {
+  const nodesRef = useRef<Array<astNode>>([])
+  /*
+    gotoLineDisableRef is used to temporarily disable the update of the view.
+    e.g when the user ask the component to "gotoLine" we don't want to rerender the component (but just to put the mouse on the desired line)
+  */
+  const referencesRef = useRef([])
+  const activeHighlightsRef = useRef([])
+  const currentNodeRef = useRef(null as nullableAstNode)
+  const gasEstimationRef = useRef({} as gasEstimationType)
+  const gotoLineDisableRef = useRef(false)
+  const [, setNode] = useState<Array<astNode>>([])
```

Comment:
Why is `setNode` needed when the value is not being used?

Reply:
yes I m trying to improve that now

Reply:
should be better now